### ISSUE TITLE
os-/cpu-independent targets

### ISSUE THREAD

**Issue by novas0x2a** [theoremlp/rules_multitool#66]
We've been integrating multitool, and we've realized that (due to the :cwd target) it's kind of convenient to be able to register other types of things with multitool, too, which keeps our tool setup nice and consistent. Today, it seems like the `os` and `cpu` fields are mandatory in the spec, but I was wondering what you thought about being able to pin multiple types in one field, for example:
```json
...
  "git-clang-format": {
    "binaries": [
      {
        "kind": "file",
        "url": "https://raw.githubusercontent.com/my-shell-script.sh",
        "sha256": "abc123",
      },
...
```

(dropping os/cpu completely, and having it be the equivalent of "all", or alternately, something like this:

```json
...
  "git-clang-format": {
    "binaries": [
      {
        "kind": "file",
        "url": "https://raw.githubusercontent.com/my-shell-script.sh",
        "sha256": "abc123",
        "platforms": [
            {"os": "linux"}, # also permits a cpu field
            {"os": "darwin"}
        ],
      },
      {
        "kind": "file",
        "url": "https://raw.githubusercontent.com/my-powershell-script.ps1",
        "sha256": "abc123",
        "platforms": [
            {"os": "windows"}
        ],
      },
...
```
In other words, a new platform key which allows one record to support multiple plats, while still being able to express differences.

**Comment by mark-thm** [comment]
I think the way I’d recommend solving this is to generate your lockfile from an input that allows specifying the tools you want in whatever syntax makes sense to you. 

Separately, the cwd rule is available in the public part of the ruleset API.
